include/game.h

```diff
@@ -31,6 +31,7 @@
 #endif
 
 #define SPAD(x) ((s32*)SP(x * sizeof(s32)))
+#define SP2(x, type) ((type*)SP(x * sizeof(void*)))
```

Comment:
I am not a fan of the introduction of this helper. The file `src/st/st0/3C5C0.c` has an interesting interpretation of how a structured stackpad could be read. Essentially you define a struct and then you leverage `offsetof` to accurately get back the offset of the field you're interested on. This comes at the sacrifice that you would have to explicitly cast the various `(Point16*)`, `(SVECTOR*)` and so on, but it guarantees the code is moddable and portable.